### PR TITLE
Add instructions to change sbt settings based on scala binary version

### PR DESCRIPTION
**Pull request by luca-battistelli**

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @luca-battistelli,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by luca-battistelli** [comment]
Recently run into this issue and spent too much time figuring it out, would be nice to have it in the official docs

File: src/reference/02-DetailTopics/01-Using-sbt/03-Cross-Build.md [hunk 1]
```diff
@@ -308,6 +308,32 @@ A custom function is mainly used when cross-building and a dependency
 isn't available for all Scala versions or it uses a different convention
 than the default.
 
+#### Conditional sbt statements depending on binary version
+
+Sometimes you might need to change your sbt settings at runtime, depending on which binary
```
Comment (eed3si9n) [inline src/reference/02-DetailTopics/01-Using-sbt/03-Cross-Build.md:313]: This statement doesn't make sense. What setting is being changed at what program's runtime?

File: src/reference/02-DetailTopics/01-Using-sbt/03-Cross-Build.md [hunk 2]
```diff
@@ -308,6 +308,32 @@ A custom function is mainly used when cross-building and a dependency
...
+  }
+}.value
+
+libraryDependencies ++= {
+  (ThisBuild / scalaVersion) {
+    case v if v startsWith "2.12" => Seq(compilerPlugin("org.scalamacros" % "paradise_2.12.10" % "2.1.0"))
+    case v if v startsWith "2.13" => Seq[ModuleID]()
+  }
+}.value
+```
```
Comment (eed3si9n) [inline src/reference/02-DetailTopics/01-Using-sbt/03-Cross-Build.md:335]: I think this is overly complicated... I can send another PR that does the same thing.

**Comment by eed3si9n** [comment]
Thanks for getting this started. I'm closing this in favor of https://github.com/sbt/website/pull/918.

**Pull request closed without merging**